server/csrf/skipper.go

```diff
@@ -0,0 +1,31 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package csrf
+
+import (
+	"github.com/labstack/echo/v4"
+)
+
+func SkipOnAuthorizationHeader(c echo.Context) bool {
+	return c.Request().Header.Get(echo.HeaderAuthorization) != ""
```

Comment:
This looks like it'll skip csrf protection if the authorization header is set to any value. If that's the case, then it allows a full bypass of CSRF protection with an invalid authorization header.

Reply:
Since CSRF attacks use the implicit sending of cookies shouldn't this be okay because a user wouldn't be able to add this header to bypass the CSRF check AND successfully become authenticated?

Reply:
correct, if authorization header is set (though since this is an explicit auth flow attackers shouldn't be able to do that with the CSRF attack) then it's up to the authorization plugin on Temporal side to deny the request

Reply:
as discussed directly adding unit tests to make sure `authorization` header is always prioritized over an auth cookie here https://github.com/temporalio/ui-server/blob/e9c7a9ceeb1fe60aac9ee190e126ae0c5415cc3a/server/auth/auth.go#L148